src/libpgmoneta/wal.c

```diff
@@ -625,6 +641,59 @@ pgmoneta_wal(int srv, char** argv)
    exit(1);
 }
 
+static int
+wal_read_replication_slot(SSL* ssl, int socket, char* slot, char* name, uint32_t* high32, uint32_t* low32, uint32_t* timeline, int segsize)
+{
+   struct message* read_slot_msg = NULL;
+   struct query_response* read_slot_response = NULL;
+   char* lsn = NULL;
+   int status = 0;
+   uint32_t local_timeline = 0;
+
+   pgmoneta_create_read_replication_slot_message(slot, &read_slot_msg);
+   status = pgmoneta_query_execute(ssl, socket, read_slot_msg, &read_slot_response);
+
+   if (status != 0)
+   {
+      pgmoneta_log_error("Error occurred when executing READ_REPLICATION_SLOT for slot %s on server %s", slot, name);
+      goto error;
+   }
+
+   if (read_slot_response->number_of_columns < 3)
+   {
+      pgmoneta_log_error("Invalid response from READ_REPLICATION_SLOT for slot %s on server %s", slot, name);
+      goto error;
+   }
+
+   local_timeline = atoi(pgmoneta_query_response_get_data(read_slot_response, 2));
+   if (local_timeline < 1)
+   {
+      pgmoneta_log_error("Error occurred when reading replication slot on server %s: timeline should at least be 1, but getting %d", name, local_timeline);
+      goto error;
+   }
+
+   lsn = pgmoneta_query_response_get_data(read_slot_response, 1);
+   if (wal_convert_xlogpos(lsn, high32, low32, segsize))
+   {
+      pgmoneta_log_error("Failed to convert LSN from replication slot %s on server %s", slot, name);
+      goto error;
+   }
+
+   *timeline = local_timeline;
+
+   pgmoneta_log_info("Success read replica: timeline %d, LSN %s", *timeline, lsn);
```

Comment:
This is debug, and replication slot - not replica

Reply:
Sorry, this is a dirty log, and I've removed it.